tests/test_model_manager.py

```diff
@@ -0,0 +1,294 @@
+# Copyright (c) 2021 NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+from .common import test_result_collector as trc
+
+from .mocks.mock_config import MockConfig
+from .mocks.mock_os import MockOSMethods
+from .mocks.mock_model_config import MockModelConfig
+from .mocks.mock_client import MockTritonClientMethods
+
+from model_analyzer.cli.cli import CLI
+from model_analyzer.config.input.config_command_profile import ConfigCommandProfile
+from model_analyzer.config.run.run_search import RunSearch
+from model_analyzer.config.run.run_config_generator import RunConfigGenerator
+from model_analyzer.constants import LOGGER_NAME
+from model_analyzer.model_manager import ModelManager
+from model_analyzer.state.analyzer_state_manager import AnalyzerStateManager
+
+import itertools
+import logging
+import sys
+
+from unittest.mock import MagicMock
+from unittest.mock import patch
+
+
+
+class ModelManagerSubclass(ModelManager):
```

Comment:
this should be renamed to `MockModelManager` and be put in `mock_model_manager.py` in `mocks` directory

Reply:
This isn't a generic mocked ModelManager that should be used by any other tests. This is very specifically part of testing in this file.